### DISCUSSION TITLE
How to manage component containing more than one form input

### DISCUSSION THREAD

**Discussion by steveoh** [react-hook-form/react-hook-form#6666] [Q&A]
I have a custom element that is either a textarea or a file input. It handles hiding the other input when the other has a value. 

Controller looks to manage one form input at a time. How would you recommend managing a component that is more than one form input?

**Comment by bluebill1049** [comment]
I would build your custom component with the logic internally and wrapped with Controller to just collect data and validate.

```tsx
const { type } = watch();

const Custm  = ({ value, onChage, type }) => {
  if (type === '') return <textarae {...props} />;
  if (type === '') return <input {...props} />;
}

<Controller render={({ field }) => <Custom {...field} type={type} />
```

**Reply by steveoh** [reply]
so the component returns the textarea *and* the input. Then once either input has a value, the other input is removed.

So the ref and name that are passed in cannot be shared on both elements.

Right now i'm trying conditionally spreading the field provided by RHF onto the input that has been interacted with

**Reply by bluebill1049** [reply]
So u only want to submit dirty fields?

**Reply by steveoh** [reply]
I am treating the two inputs as one. I just pushed some code that is working but i'm not sure that it is ideal and I hope you can point out some places where it can be improved. 

This is where I am creating the custom component and I am passing in quite a few react hook form properties.

https://github.com/uic-utah/uic-inventory/blob/feat/well-descriptions/src/api/uic-inventory/src/components/Pages/Wells/AddWellDetails.jsx#L334-L354

This is the component https://github.com/uic-utah/uic-inventory/blob/feat/well-descriptions/src/api/uic-inventory/src/components/FormElements/LimitedTextarea.jsx#L127

I understand if this is beyond the kind of effort you are interested in providing here but I sure would appreciate any type of review that you could provide of my usage.

Here's a quick video of the form in use

https://user-images.githubusercontent.com/325813/135549537-600cacf5-05fc-485e-b628-811d8dd7572c.mp4
